core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerRequestCompletionHandler.scala

```diff
@@ -61,10 +62,10 @@ class TransactionMarkerRequestCompletionHandler(transactionMarkerChannel: Transa
         for ((topicPartition: TopicPartition, error: Errors) <- errors) {
           error match {
             case Errors.NONE =>
-              transactionMarkerChannel.pendingTxnMetadata(epochAndMarkers.metadataPartition, txnMarker.producerId()) match {
+              transactionMarkerChannel.pendingTxnMetadata(metadataPartition, txnMarker.producerId()) match {
```

Comment:
Could we access the `transactionMetadata` by sharing the cache map from `txnStateManager` directly?